deployinggpucontainers.mdx

```diff
@@ -0,0 +1,95 @@
+---
+title: 'Deploying GPU Containers'
+---
+### Background
+Accessing affordable GPU compute often gets you out of flow. Big clouds have complicated UI's and high prices, but 3rd party clouds have no supply guarantees.
+Now with the Shadeform API, you can spin up GPU containers on the most affordable compute without leaving your IDE.
+
+### Benchmarking Models
+
+With the Container Launch config, we can spin up a secure VM, and start a container on it automatically. Let's start with a very simple example - running evals on a new model from Huggingface. 
+
+We have an image that does this with the LM Eval Harness library using HF Transformers. The code for it can be found and forked [here](https://github.com/shadeform/examples/tree/main/lm-eval-harness-benchmarking).
+We also have a [notebook](https://github.com/shadeform/examples/blob/main/deploy_container.ipynb) that automatically launches this image on the most affordable A6000's on the market to run the 
+[Mistral-7b-Instruct-v0.2](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.2) model though a benchmark. You can configure the model, gpu type, number of fewshot examples, and the benchmark eval.
+
+### Setup
+
+The requirements are simple, so in your preferred python environment:
+
+```bash
+git clone https://github.com/shadeform/examples.git
+cd examples/
+```
+Then in `deploy_container.ipynb` input your Shadeform API Key from [here](https://platform.shadeform.ai/settings/api), and a  `huggingface_token` if the model you need is gated (like Llama2 or Gemma).
+
+The notebook kicks off a VM request and runs the container specifically on a 1xA6000 Machine which are quite afforable on our platform. The container will download the model and run the benchmark for us. 
+
+The notebook leverages a lot of the principles found in [this post](https://docs.shadeform.ai/findingmostaffordablegpus) for using the platform to _find_ affordable GPU's.
+
+### Deploying a container
+
+Once we have an available instance, we will deploy a container that runs [lm-eval-harness](https://github.com/EleutherAI/lm-evaluation-harness/tree/main) on the model.
+Here is where we pass in environment variables, and select the docker image via our API.
+
+```python
+payload = {
+  "cloud": best_instance["cloud"],
+  "region": region,
+  "shade_instance_type": shade_instance_type,
+  "shade_cloud": True,
+  "name": "cool_gpu_server",
+  "launch_configuration": {
+    "type": "docker",
+    "docker_configuration": {
+      "image": "shadeform/lm-eval-harness",
+      "envs": [
```

Comment:
Can we add a line or two describing that this is passing environment variables into the container?

Reply:
yes, will also put this into the notebook source as well